artifactory/services/go/go.go

```diff
@@ -32,12 +32,11 @@ func (gs *GoService) PublishPackage(params GoParams) (*utils.OperationSummary, e
 	if err != nil {
 		return nil, err
 	}
-	publisher := GetCompatiblePublisher(artifactoryVersion)
-	if publisher == nil {
+	publisher := &GoPublishCommand{}
+	if !publisher.isCompatible(artifactoryVersion) {
+		// PublishPackage support Artifactory version "6.6.1" and above.
 		return nil, errorutils.CheckError(errors.New(fmt.Sprintf("Unsupported version of Artifactory: %s", artifactoryVersion)))
```

Comment:
1. Let's also add the minimum supported Artifactory version to this error message.
2. To avoid duplicating the the min supported version in multiple place in the code, you can rename the  ```isCompatible``` function to ```verifyCompatibleVersion``` and have it return this error, with the message, instead of a boolean.